flink-runtime/src/main/java/org/apache/flink/runtime/clusterframework/types/ResourceProfile.java

```diff
@@ -485,15 +486,11 @@ public String toString() {
             return "ResourceProfile{ANY}";
         }
 
-        final StringBuilder extendedResourceStr = new StringBuilder(extendedResources.size() * 10);
-        for (Map.Entry<String, ExternalResource> resource : extendedResources.entrySet()) {
-            extendedResourceStr
-                    .append(", ")
-                    .append(resource.getKey())
-                    .append('=')
-                    .append(resource.getValue().getValue());
-        }
-        return "ResourceProfile{" + getResourceString() + extendedResourceStr + '}';
+        return "ResourceProfile{"
+                + getResourceString()
+                + ", "
+                + ExternalResourceUtils.generateExternalResourcesString(extendedResources)
```

Comment:
What if there's no external resource? We will have something like `ResourceProfile{a=b, c=d, }`.

Reply:
Same for `TaskExecutorProcessSpec` and `WorkerResourceSpec`.